src/contracts/EthBridge.h

```diff
@@ -0,0 +1,634 @@
+                '\0'
+            };
+            LOG_INFO(locals.log);
+            output.status = 1; // Error
+            return;
+        }
+
+        // Check the status
+        if (order.status != 0) { // Ensure it's not already completed or refunded
+            locals.log = EthBridgeLogger{
+                CONTRACT_INDEX,
+                EthBridgeError::invalidOrderState,
+                input.orderId,
+                0,
+                '\0'
+            };
+            LOG_INFO(locals.log);
+            output.status = 2; // Error
+            return;
+        }
+
+        // Handle order based on transfer direction
+        if (order.fromQubicToEthereum) {
+            // Ensure sufficient tokens were transferred to the contract
+            if (state.totalReceivedTokens < order.amount) {
+                locals.log = EthBridgeLogger{
+                    CONTRACT_INDEX,
+                    EthBridgeError::insufficientLockedTokens,
+                    input.orderId,
+                    order.amount,
+                    '\0'
+                };
+                LOG_INFO(locals.log);
+                output.status = 4; // Error
+                return;
+            }
+
+            state.lockedTokens += order.amount;
+        }
+        else {
+            // Ensure sufficient tokens are locked for the order
+            if (state.lockedTokens < order.amount) {
+                locals.log = EthBridgeLogger{
+                    CONTRACT_INDEX,
+                    EthBridgeError::insufficientLockedTokens,
+                    input.orderId,
+                    order.amount,
+                    '\0'
+                };
+                LOG_INFO(locals.log);
+                output.status = 5; // Error
+                return;
+            }
+
+            // Transfer tokens back to the user
+            if (qpi.transfer(order.qubicSender, order.amount) < 0) {
+                locals.log = EthBridgeLogger{
+                    CONTRACT_INDEX,
+                    EthBridgeError::transferFailed,
+                    input.orderId,
+                    order.amount,
+                    '\0'
+                };
+                LOG_INFO(locals.log);
+                output.status = 6; // Error
+                return;
+            }
+
+            state.lockedTokens -= order.amount;
+        }
+
+        // Mark the order as completed
+        order.status = 1; // Completed
+        state.orders.set(order.orderId, order);
+
+        output.status = 0; // Success
+        locals.log = EthBridgeLogger{
+            CONTRACT_INDEX,
+            0, // No error
+            input.orderId,
+            order.amount,
+            '\0'
+        };
+        LOG_INFO(locals.log);
+    _
+
+    // Refund an order and unlock tokens
+    struct refundOrder_locals {
+        EthBridgeLogger log;
+    };
+
+    PUBLIC_PROCEDURE_WITH_LOCALS(refundOrder)
+        id invocatorAddress = qpi.invocator();
+        bit isManagerOperating = false;
```

Comment:
This should be in the `locals` because we don't allow local variables

Reply:
Oh, true! Already changed!